tools/release/release.sh

```diff
@@ -0,0 +1,318 @@
+    printf "Fatal: ${buildToolsPath} folder not found, ensure that you have installed the SDK version ${buildToolsVersion}.\n"
+    exit 1
+fi
+
+# Check if git flow is enabled
+git flow config >/dev/null 2>&1
+if [[ $? == 0 ]]
+then
+    printf "Git flow is initialized\n"
+else
+    printf "Git flow is not initialized. Initializing...\n"
+    # All default value, just set 'v' for tag prefix
+    git flow init -d -t 'v'
+fi
+
+printf "OK\n"
+
+printf "\n================================================================================\n"
+printf "Ensuring main and develop branches are up to date...\n"
+
+git checkout main
+git pull
+git checkout develop
+git pull
+
+printf "\n================================================================================\n"
+# Guessing version to propose a default version
+versionsFile="./plugins/src/main/kotlin/Versions.kt"
+versionMajorCandidate=`grep "val versionMajor" ${versionsFile} | cut  -d " " -f6`
+versionMinorCandidate=`grep "val versionMinor" ${versionsFile} | cut  -d " " -f6`
+versionPatchCandidate=`grep "val versionPatch" ${versionsFile} | cut  -d " " -f6`
+versionCandidate="${versionMajorCandidate}.${versionMinorCandidate}.${versionPatchCandidate}"
+
+read -p "Please enter the release version (example: ${versionCandidate}). Just press enter if ${versionCandidate} is correct. " version
+version=${version:-${versionCandidate}}
+
+# extract major, minor and patch for future use
+versionMajor=`echo ${version} | cut  -d "." -f1`
+versionMinor=`echo ${version} | cut  -d "." -f2`
+versionPatch=`echo ${version} | cut  -d "." -f3`
+nextPatchVersion=$((versionPatch + 2))
+
+printf "\n================================================================================\n"
+printf "Starting the release ${version}\n"
+git flow release start ${version}
+
+# Note: in case the release is already started and the script is started again, checkout the release branch again.
+ret=$?
+if [[ $ret -ne 0 ]]; then
+  printf "Mmh, it seems that the release is already started. Checking out the release branch...\n"
+  git checkout "release/${version}"
+fi
+
+# Ensure version is OK
+versionsFileBak="${versionsFile}.bak"
+cp ${versionsFile} ${versionsFileBak}
+sed "s/private const val versionMajor = .*/private const val versionMajor = ${versionMajor}/" ${versionsFileBak} > ${versionsFile}
+sed "s/private const val versionMinor = .*/private const val versionMinor = ${versionMinor}/" ${versionsFile}    > ${versionsFileBak}
+sed "s/private const val versionPatch = .*/private const val versionPatch = ${versionPatch}/" ${versionsFileBak} > ${versionsFile}
+rm ${versionsFileBak}
+
+# This commit may have no effect because generally we do not change the version during the release.
+git commit -a -m "Setting version for the release ${version}"
+
+printf "\n================================================================================\n"
+printf "Building the bundle locally first...\n"
+./gradlew clean bundleRelease
```

Comment:
I'm getting OOM errors while building. Maybe we should bump the heap size, either globally or just for this command? Also, the result of this step is not checked before continuing.

Reply:
> I'm getting OOM errors while building

I probably have a local config somewhere which avoid that. The idea of this line is to detect in advance if the CI will fail. I will see how to fix this issue, but since I do not encounter it, maybe you can propose a fix?

> the result of this step is not checked before continuing

Any failure should break the script now (there is no `set +e` statement), so I think it's fine, but I can double check.

Reply:
> maybe you can propose a fix?

The CI is actually using 3GB of RAM (`-Xmx3072m`) while we're using 2GB locally (`-Xmx2048m`), so we could increase that. It would also be great if we could override this value locally, but I couldn't find an easy way to do it when I last checked.

Reply:
I tried again and it built fine using just the default 2GB now. Then I decreased to 1GB to force an OOM error and the script continues after gradle fails:

```
* What went wrong:
Execution failed for task ':app:signReleaseBundle'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.FinalizeBundleTask$BundleToolRunnable
   > java.lang.OutOfMemoryError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 37s
1868 actionable tasks: 552 executed, 1039 from cache, 277 up-to-date

================================================================================
Running towncrier...
Loading template...
Finding news fragments...
Rendering news fragments...
Writing to newsfile...
Staging newsfile...
Removing news fragments...
I want to remove the following files:
Is it okay if I remove those files? [Y/n]: Done!

================================================================================
Check the file CHANGES.md consistency. It's possible to reorder items (most important changes first) or change their section if relevant. Also an opportunity to fix some typo, or rewrite things. Do not commit your change. Press enter when it's done.
```